nomad/node_endpoint.go

```diff
@@ -1220,6 +1220,7 @@ func (n *Node) UpdateAlloc(args *structs.AllocUpdateRequest, reply *structs.Gene
 		if evalTriggerBy != structs.EvalTriggerJobDeregister &&
 			alloc.ClientStatus == structs.AllocClientStatusUnknown {
 			evalTriggerBy = structs.EvalTriggerReconnect
+			alloc.ReconnectModifyIndex = allocToUpdate.ReconnectModifyIndex
```

Comment:
This assignment corrupts the state store because `alloc` hasn't been copied after being queried from the state store. I'm fairly certain this line isn't needed at all, as the `allocToUpdate` is what's getting added to the batch of updates and not `alloc`.